### PR TITLE
HIVE-24561: Deflake TestCachedStoreUpdateUsingEvents test

### PR DESCRIPTION
**Pull request by vihangk1**
### What changes were proposed in this pull request?
This PR modifies the test TestCachedStoreUpdateUsingEvents to not use "file:/tmp" which is error prone since clean up method can delete all the files in /tmp. Also fixes a harmless NPE which unnecessarily spams the log file.

### Why are the changes needed?
I observed this test to be failing on a unrelated PR. I am hoping that these changes will help remove any flakiness if present.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Ran the test locally.

### CODE REVIEW

**Review by yongzhi — APPROVED** [review]
LGTM

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/cache/TestCachedStoreUpdateUsingEvents.java [hunk 1]
```diff
@@ -82,9 +84,9 @@ public void setUp() throws Exception {
     HiveMetaStore.HMSHandler.createDefaultCatalog(rawStore, new Warehouse(conf));
   }
 
-  private Database createTestDb(String dbName, String dbOwner) {
+  private Database createTestDb(String dbName, String dbOwner) throws IOException {
     String dbDescription = dbName;
-    String dbLocation = "file:/tmp";
+    String dbLocation = Files.createTempDirectory(dbName).toString();
```
Comment (kishendas) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/cache/TestCachedStoreUpdateUsingEvents.java:89]: Even the dbName can be same across multiple tests, which can get cleaned up. Would it be a good idea to use the current timestamp or generate a random number and use that for the temporary directory ?

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/cache/TestCachedStoreUpdateUsingEvents.java [hunk 2]
```diff
@@ -94,8 +96,9 @@ private Database createTestDb(String dbName, String dbOwner) {
   }
 
   private Table createTestTblParam(String dbName, String tblName, String tblOwner,
-                              List<FieldSchema> cols, List<FieldSchema> ptnCols, Map<String, String> tblParams) {
-    String serdeLocation = "file:/tmp";
+                              List<FieldSchema> cols, List<FieldSchema> ptnCols, Map<String, String> tblParams)
+      throws IOException {
+    String serdeLocation = Files.createTempDirectory(dbName + "_" + tblName).toString();
```
Comment (kishendas) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/cache/TestCachedStoreUpdateUsingEvents.java:101]: Even the dbName and tblName can be same across multiple tests, which can get cleaned up. Would it be a good idea to use the current timestamp or generate a random number and use that for the temporary directory ?

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/cache/TestCachedStoreUpdateUsingEvents.java (hunk 1 above)
Comment (vihangk1) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/cache/TestCachedStoreUpdateUsingEvents.java:89]: yeah, the Files.createTempDirectory creates a temp directory by appending the timestamp. DbName is just used as a prefix for the directory name.

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/cache/TestCachedStoreUpdateUsingEvents.java (hunk 2 above)
Comment (vihangk1) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/metastore/cache/TestCachedStoreUpdateUsingEvents.java:101]: same comment as above. Based on the documentation of Files.createTempDirectory it returns "the path to the newly created directory that did not exist before this method was invoked"

**Review by kishendas — APPROVED** [review]

**Pull request merged**